### DISCUSSION TITLE
Customizing views, layout, css

### DISCUSSION THREAD

**Discussion by mauriceb77** [codeigniter4/shield#470] [Q&A]
Hello,

What is the best way to customize the register and login views? I copied them to my own view files under App\Views but the views use the Shield layout view (header, css, etc.). Should I just use my own layout view and update the Auth config? I think this is best approach but want to make sure I am following best practice.

Thanks,
-mb

**Comment by datamweb** [comment]
Edit file `app\Config\Auth.php` and replace the new paths:
```shell
    public array $views = [
        'login'                       => '\App\Views\Shield\login',
        'register'                    => '\CodeIgniter\Shield\Views\register',
        'layout'                      => '\CodeIgniter\Shield\Views\layout',
        'action_email_2fa'            => '\CodeIgniter\Shield\Views\email_2fa_show',
        'action_email_2fa_verify'     => '\CodeIgniter\Shield\Views\email_2fa_verify',
        'action_email_2fa_email'      => '\App\Views\Shield\Email\email_2fa_email',
        'action_email_activate_show'  => '\CodeIgniter\Shield\Views\email_activate_show',
        'action_email_activate_email' => '\CodeIgniter\Shield\Views\Email\email_activate_email',
        'magic-link-login'            => '\CodeIgniter\Shield\Views\magic_link_form',
        'magic-link-message'          => '\CodeIgniter\Shield\Views\magic_link_message',
        'magic-link-email'            => '\CodeIgniter\Shield\Views\Email\magic_link_email',
    ];
```

**Reply by mauriceb77** [reply]
Thanks, that was my approach, just making sure. 

I am modifying the registration view but need to include some fields which require an ajax call to a custom class. I need to extend the Shield RegisterController so I can include my model, etc., correct?
